### PR TITLE
Update gdscript queries

### PR DESCRIPTION
**Pull request by PrestonKnopp**
Updates highlights, locals, and indents and adds folds. Also add myself as maintainer.

### CODE REVIEW

File: queries/gdscript/highlights.scm
Comment (amaanq) [inline queries/gdscript/highlights.scm:10]: Why are these `@symbol`? For node_path, if it's a location/uri-like string, maybe `@text.uri`?

File: queries/gdscript/highlights.scm
Comment (amaanq) [inline queries/gdscript/highlights.scm:32]: Are strings here actually comments?

File: queries/gdscript/highlights.scm
Comment (amaanq) [inline queries/gdscript/highlights.scm:161]: Just a note for #4382, will move this to `@keyword.coroutine` once this is merged

File: queries/gdscript/highlights.scm
Comment (amaanq) [inline queries/gdscript/highlights.scm:166]: Would yield also apply to `@keyword.coroutine`?

File: queries/gdscript/highlights.scm
Comment (amaanq) [inline queries/gdscript/highlights.scm:189]: Are all annotation identifiers not `@attribute`? Is the annotation itself not entirely `@attribute` as well? I'll play around with these changes shortly..

File: queries/gdscript/indents.scm
Comment (amaanq) [inline queries/gdscript/indents.scm:28]:
```suggestion
[
 (string)
 (comment)
 (array)
 (dictionary)
 (parenthesized_expression)
 (ERROR)
] @auto
```

**Review by amaanq — CHANGES_REQUESTED** [review]

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:10]: Ah yes, `@text.uri` fits better than `@symbol`.

`get_node` is sugar for getting a ref to a node in godot's "scene tree" (`$get/this/node`). It's symbol like I would say. What would you suggest?

And now that I see it `string_name` should actually just be `@string`.

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:32]: Yes, strings as standalone statements are valid comments. Here's the PR to verify: https://github.com/godotengine/godot/pull/72971

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:166]: Yes it would apply

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:189]: I was thinking about that. There's only a handful of annotations, you can't define your own, and they are hard coded in Godot's Gdscript parser. Maybe the highlighting can show when you have a typo at glance?

> Is the annotation itself not entirely @attribute as well?

I followed queries from java: https://github.com/nvim-treesitter/nvim-treesitter/blob/3fe80dbecdf1fa88cf81a4d7b30ab0714f8c443a/queries/java/highlights.scm#L37-L40

and python: https://github.com/nvim-treesitter/nvim-treesitter/blob/3fe80dbecdf1fa88cf81a4d7b30ab0714f8c443a/queries/python/highlights.scm#L83-L87

... which I see now that python does "@" with a priority.

I'll mark the whole annotation as `@attribute`.

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:10]: Great suggestion, highlight support for `@text.uri` looks really good on both `get_node` and `node_path`.

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:164]: I have subscribed to #4382.

File: queries/gdscript/highlights.scm
Comment (amaanq) [reply queries/gdscript/highlights.scm]: No, no tool should really be using that to determine links/paths to open, and I don't know of any. I'll leave this one up to you though, since the `$` beforehand is a bit funky but I'm not familiar with godot at all.

File: queries/gdscript/highlights.scm
Comment (amaanq) [reply queries/gdscript/highlights.scm]: While it's good to gain understanding and learn the standard from other queries, it's best not to think of it as a "one size fits all" when there are rules with similar names/functionality :)

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm]: Okay, I'll keep the change to `@text.uri`.

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm]: That's good to know that the queries should be tailored specifically for the language. I added `@attribute` to the `"@"` and kept the any-of check.

**Comment by clason** [comment]
Please don't merge master into a PR branch; it's better to rebase to keep history linear.

(Otherwise we'll just squash all your commits down to one.)

**Comment by PrestonKnopp** [comment]
My bad, I didn't know the sync would merge rather than rebase.

File: queries/gdscript/highlights.scm
Comment (amaanq) [inline queries/gdscript/highlights.scm:88]: what is pattern_open_ending? and is underscore used in a pinctuation-like manner?

File: queries/gdscript/highlights.scm
Comment (amaanq) [inline queries/gdscript/highlights.scm:8]: string_name is a confusing name, can you provide more context/info as to what it is?

**Review by amaanq — CHANGES_REQUESTED** [review]

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:88]: Good catch, underscore should be renamed to `pattern_wildcard` in the grammar. Here's example usage of the two:

```gdscript
match some_value:
    ["open ended array pattern", ..]: pass
    ["match this", _, "and this"]: pass
    _: print("Match anything here")
```

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:8]: Sure, it's syntax sugar for Godot's concept of interned strings. They look like:

```gdscript
&"interned string"
```

File: queries/gdscript/highlights.scm
Comment (theHamsta) [inline queries/gdscript/highlights.scm:82]: constructor is not for the keyword new but for the type identifier to be constructed. C++ uses `@keyword.operator` for `new`.

**Review by theHamsta — APPROVED** [review]

File: queries/gdscript/highlights.scm
Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm]: Done. Also realized (attribute (identifier)) still matches all attribute calls so removed that nesting.

File: queries/gdscript/highlights.scm
Comment (amaanq) [reply queries/gdscript/highlights.scm:88]: Go has `_` under `@constant`, and `..` under `@operator`, does that fit better here? Seems to be used similarly

Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm:88]: Yeah, those sound good to me.

Comment (PrestonKnopp) [reply queries/gdscript/highlights.scm]: Done.

**Comment by amaanq** [comment]
Thanks!

**Review by amaanq — APPROVED** [review]

**Pull request merged**
